### ISSUE TITLE
Incorrect diagnostic for static readonly field

### ISSUE THREAD

**Issue by GeorgeAlexandria** [JosefPihrt/Roslynator#434]
Hi, 

I'm used Roslyntor 2017 v1.9.2.

Roslynator suggests the incorrect code fix for a static readonly field that will be modified or will be used as `ref|out|in` argument for method:

```c#
...
class Sample
{
    private static readonly int val = 100;

    static Sample()
    {
        val = 25;
        Init(ref val);
    }

    // NOTE: example with the ref parameter will be enough
    private static void Init(ref int arg)
    {
        ...
    }
}
```

So after applying the suggestion [RCS1187](https://github.com/JosefPihrt/Roslynator/blob/master/docs/analyzers/RCS1187.md) you will receive a compile time errors.

**Comment by GeorgeAlexandria** [comment]
@JosefPihrt, I looked at the fix for this issue and I think that it didn't solve the all cases in this issue. I mean case when your pass field as `in` argument to a method. In the `AssignedExpressionWalker` you only [check](https://github.com/JosefPihrt/Roslynator/blob/943815f0c5ec3bf99e6a96b2d99f687e63f47fcf/src/CSharp/CSharp/SyntaxWalkers/AssignedExpressionWalker.cs#L73) that this argument is `ref` or `out`, but not `in`. So you still suggest incorrect diagnostics, but only for `in` argument.

Let me know if I missing something.

**Comment by JosefPihrt** [comment]
@GeorgeAlexandria you are right, thanks for noticing that.
